[P1-N07] Contract naming updates

chrismaree opened the pull request:
This PR re-names a number of internal variables and functions to be more descriptive and consistent. Most changes are small and internal variables. A number of changes requested by OZ have been left out as these will have ripple effects throughout the code base and it was decided that the value add in make these changes is less than the cost of updating other parts of the infrastructure.

The following changes were made:
- In the Withdrawable contract The `owner` parameter of the `createWithdrawRole` was be renamed to `withdrawerAddress`.
- In the Store contract The `fixedOracleFeePerSecond` and `weeklyDelayFee` variables were renamed to `fixedOracleFeePerSecondPerPfc` and `weeklyDelayFeePerPfc`. This includes all other references to fees like events.
- In the `FixedPoint` contract the `divRaw` variable was renamed to `aScaled` since it has nothing to do with division.

The following changes recommended by OZ were not implemented:
- All changes to `multiRole` as this will have a large ripple effect throughout the code base.
- Renaming `EncryptedStore` as this is removed in a later PR.
- `Governor` change  `_uintToBytes` to `_uintToBytes32` as this is addressed in another PR.
- Changing `getPendingRequests()` to `getActiveRequests()` as this has a large ripple effect for minimum benefit.

nicholaspai commented on core/contracts/common/implementation/MultiRole.sol:
```diff
@@ -10,17 +10,17 @@ library Exclusive {
     }
 
-    function resetMember(RoleMembership storage roleMembership, address newMember) internal {
+    function resetExclusiveMember(RoleMembership storage roleMembership, address newMember) internal {
```
This is called in `scripts/TransferPermissions`

chrismaree replied:
I don't think it is. the `resetMember` function that is called in that script is from the `abstract contract MultiRole` on line [122](https://github.com/UMAprotocol/protocol/blob/5e33a7efa6955d4ee17e481822aa03dea9c7541e/core/contracts/common/implementation/MultiRole.sol#L122) and this function is not changed in this PR. Rather, this PR changes the function that `resetMember` calls within the context of the exclusiveRoleMembership.

chrismaree replied:
There is a unit test to ensure that `TransferPermissions` works correctly in  `test/scripts/TransferPermissions.js` which passed after this change.

nicholaspai commented on core/contracts/common/implementation/MultiRole.sol:
```diff
@@ -10,17 +10,17 @@ library Exclusive {
     }
 
-    function getMember(RoleMembership storage roleMembership) internal view returns (address) {
+    function getExclusiveMember(RoleMembership storage roleMembership) internal view returns (address) {
```
In `scripts/cli/wallet/getTwoKeyContract` and `scripts/TransferPermissions`

chrismaree replied:
I think this is the same as above in that the base `MultiRole` function has not changed it's name so should still be fine. Do you agree?

nicholaspai commented on core/contracts/oracle/implementation/Store.sol:
```diff
@@ -120,25 +121,28 @@ contract Store is StoreInterface, Withdrawable {
-     * @param newOracleFee new fee per second charged to use the oracle.
+     * @param newOracleFeePerSecondPerPFC new fee per second charged to use the oracle.
      */
-    function setFixedOracleFeePerSecond(FixedPoint.Unsigned memory newOracleFee)
```
In `scripts/PrecisionErrors.js` and `scripts/IntegrationTests.js`

chrismaree replied:
these changes were made in these scripts already :)

nicholaspai commented on core/contracts/oracle/implementation/Voting.sol:
```diff
@@ -536,12 +536,12 @@ contract Voting is Testable, Ownable, OracleInterface, VotingInterface, Encrypte
      */
     // TODO(#969) Remove once prettier-plugin-solidity can handle the "override" keyword
     // prettier-ignore
-    function getPendingRequests() external override view returns (PendingRequest[] memory pendingRequests) {
```
In `scripts/Voting.js`, `scripts/cli/admin.js`, `scripts/cli/voting/commitVotes`, `scripts/cli/voting/revealVotes`, `scripts/cli/voting/displayStatus` and `voter-dapp/src/ActiveRequests`

chrismaree replied:
I think I got all of these in the PR.

nicholaspai commented on core/contracts/common/implementation/MultiRole.sol:
```diff
@@ -137,7 +137,7 @@ abstract contract MultiRole {
      * managing role for `roleId`.
      */
     function addMember(uint roleId, address newMember) public onlyShared(roleId) onlyRoleManager(roleId) {
```
This is in a ton of scripts, including some migration scripts

nicholaspai commented on core/contracts/common/implementation/MultiRole.sol:
```diff
@@ -146,7 +146,7 @@ abstract contract MultiRole {
      * managing role for `roleId`.
      */
     function removeMember(uint roleId, address memberToRemove) public onlyShared(roleId) onlyRoleManager(roleId) {
```
Same as `addMember`

nicholaspai commented:
@chrismaree looks good, I noticed you didn't implement the following suggestions:
- Changing `addMember` to `addSharedMember`
- Changing `removeMember` to `removeSharedMember`
- In `Governor`: change `_uintToBytes()` to `_uintToBytes32()`

Just wanted to flag in case you didn't see, or curious if you decided it wasn't worth it. The `MultiRole` changes have a ton of upstream calls.

chrismaree commented:
> @chrismaree looks good, I noticed you didn't implement the following suggestions:
> 
> * Changing `addMember` to `addSharedMember`
> * Changing `removeMember` to `removeSharedMember`
> * In `Governor`: change `_uintToBytes()` to `_uintToBytes32()`
> 
> Just wanted to flag in case you didn't see, or curious if you decided it wasn't worth it. The `MultiRole` changes have a ton of upstream calls.

Great. So regarding those suggestions:

1 &2) `Multiroll` has two libraries in it: `Exclusive` and `Shared`. These are implemented in the abstract base contract `MultiRole`. I didn't change any implementation in the `MultiRole` contract but rather only in the libraries, as suggested in the OZ audit. This means that the upstream usage of contracts that inherit from `MultiRole` don't need to change. This means that I did implement the changes for going from `addMember` -> `addSharedMember` & `removeMember` -> `removeSharedMember` within the library but I did not change the abstract class as this has far reaching implications up stream. My understanding of the OZ audit comment on these points was they they only wanted changes at the library level, not within the abstract contract.

4) @mrice32 refactored this bad boy in another PR. [see](https://github.com/UMAprotocol/protocol/pull/1204)

mrice32 commented:
> My understanding of the OZ audit comment on these points was they they only wanted changes at the library level, not within the abstract contract.
>
My understanding was that they wanted the method names to be changed at the contract level to communicate that those methods only work for `shared` or `exclusive` roles, not both.

mrice32 commented on core/contracts/oracle/implementation/Store.sol:
```diff
@@ -120,25 +121,28 @@ contract Store is StoreInterface, Withdrawable {
-    function setWeeklyDelayFee(FixedPoint.Unsigned memory newWeeklyDelayFee) public onlyRoleHolder(uint(Roles.Owner)) {
-        weeklyDelayFee = newWeeklyDelayFee;
-        emit NewWeeklyDelayFee(newWeeklyDelayFee);
+    function setWeeklyDelayFeePerPFC(FixedPoint.Unsigned memory newWeeklyDelayFeePerPFC)
```
This is actually becoming per second, too in [this PR](https://github.com/UMAprotocol/protocol/pull/1237). Do you want to change it to `setWeeklyDelayFeePerSecondPerPfc`? Or should I change the name in the aforementioned PR? Something else?

mrice32 commented on core/contracts/oracle/implementation/MessageStore.sol:
```diff
@@ -8,7 +8,7 @@ pragma solidity ^0.6.0;
  * Only one message per topic is stored at any given time.
  */
-contract EncryptedStore {
+contract MessageStore {
```
I would remove all the changes wrt EncryptedStore since the entire contract is being removed in #1231. We can add a note in the response doc saying that we didn't make that change because the contract was removed.

chrismaree replied:
done

mrice32 commented on core/contracts/common/implementation/MultiRole.sol:
```diff
@@ -34,17 +34,17 @@ library Shared {
     }
 
-    function removeMember(RoleMembership storage roleMembership, address memberToRemove) internal {
+    function removeSharedMember(RoleMembership storage roleMembership, address memberToRemove) internal {
```
Should we get rid of all the multirole changes given that we don't want the interface to change? (I know these are internal changes, but I think they actually wanted us to change the external function names)

chrismaree replied:
ye, I've reverted all changes to multirole re our discussion. Thinks keeps things simple. no need to change more

chrismaree commented:
@mrice32 and @nicholaspai please take another look. I've reverted all changes other than the ones we've discussed. I also updated the PR description to better reflect this.

mrice32 commented on core/contracts/oracle/implementation/Store.sol:
```diff
@@ -25,8 +25,8 @@ contract Store is StoreInterface, Withdrawable {
-    FixedPoint.Unsigned public fixedOracleFeePerSecond; // Percentage of 1 E.g., .1 is 10% Oracle fee.
-    FixedPoint.Unsigned public weeklyDelayFee; // Percentage of 1 E.g., .1 is 10% weekly delay fee.
+    FixedPoint.Unsigned public fixedOracleFeePerSecondPerPfc; // Percentage of 1 E.g., .1 is 10% Oracle fee.
+    FixedPoint.Unsigned public weeklyDelayFeePerPfc; // Percentage of 1 E.g., .1 is 10% weekly delay fee.
```
nit: shouldn't this variable also be changed to `weeklyDelayFeePerSecondPerPfc`?

chrismaree replied:
🤦 I've changed this so many times, sorry this got through, I have updated accordingly.

mrice32 commented:
@chrismaree Looks like ci is magically fixed from before, but the test errors now are due to merge issues, I think.

chrismaree commented:
CI is passing and all looks good here fore me. @mrice32 or @ptare can you give a final look over before merging this?
